src/expr/src/scalar/func.rs

```diff
@@ -2865,6 +2865,19 @@ impl BinaryFunc {
             _ => None,
         }
     }
+
+    /// Returns true if the function could introduce an error on non-error inputs.
+    pub fn could_error(&self) -> bool {
+        match self {
+            BinaryFunc::Eq
```

Comment:
I think we could add all functions here where the implementation has an unconditional `Ok` result? Like the following:
```
            BinaryFunc::Eq => Ok(eager!(eq)),
```

Happy to keep this a follow-up task.

Reply:
Yeah, I didn't want to trawl through all the implementations and do this, and .. also do the differently handled (macros) unary implementations, all that stuff. We should clean up the various types and their methods / properties in a consistent style, but not going to do that in this PR. :D